### ISSUE TITLE
Accidentally clicked on show my location, no way to return to previous location

### ISSUE THREAD

**Issue by boothym** [openstreetmap/iD#8474]
I was editing a fair distance away from my home on the map, when I accidentally clicked on the "show my location" button (I was aiming for the background settings button). This jumped to my current location, but there was no easy way to get back to where I was editing. 

I hadn't edited anything at that point (otherwise I could click save then click on something I'd edited), so I basically had to press the back button in the browser then go back in and edit again (also easier than trying to search for my previous location).

My suggestions would be:

Add a confirmation step before jumping to the user's current location and/or 

Move the show my location button to somewhere that it won't be clicked accidentally.

**Comment by willemarcel** [comment]
> I hadn't edited anything at that point (otherwise I could click save then click on something I'd edited), so I basically had to press the back button in the browser then go back in and edit again (also easier than trying to search for my previous location).

An easy way to go back to the place you were editing is to press the `Save` button and then click on some of the features listed on the sidebar or press the Undo and Redo buttons. But I agree with you that the position of the `Show my location` is not the best one.

**Comment by 1ec5** [comment]
I often accidentally press the hotkey for this command. To prevent it from interfering with my mapping, I set my browser to always ask me for permission when openstreetmap.org asks for my location. Then I can simply decline and keep mapping.

**Comment by boothym** [comment]
@willemarcel Yes, I would have done that but like I said - I hadn't edited anything yet so I couldn't 😉 

@1ec5 I thought I had it set to "ask" but it turns out I must've set it to allow - either in iD or perhaps on OSM.org.

**Comment by 1ec5** [comment]
As reported in #8534, it’s also quite easy to accidentally invoke this command using the keyboard, particularly if you accidentally click on the map while typing something that contains the letter “j”. Compared to other single-key shortcuts like squaring corners or zooming to fit, this shortcut is enabled even when nothing is selected, and there’s no straightforward way to undo it.

For someone using the keyboard, popping up a JavaScript prompt could be problematic, because it could automatically dismiss the browser’s built-in confirmation dialog to enable geolocation. (I’m thinking of Firefox’s “doorhanger” notification UI.) I’m not sure if we can rely on a custom dialog box to get keyboard focus after the built-in confirmation dialog is dismissed, so that could be inconvenient for keyboard users.

I’d favor adding a modifier key such as <kbd>Ctrl</kbd> (Command on macOS), to make the keyboard shortcut much harder to press by accident. Maybe we could also move the button from the right strip to the blank part of the left sidebar and show it only when there’s no selection or search query. There, the ample padding could minimize the potential for misclicks.

**Comment by boothym** [comment]
@tyrasd you've closed this with a PR that removes the L key shortcut, can I just confirm that it also removes or alters the position of the location button? As that's what this issue is about.

**Comment by k-yle** [comment]
yeah this should not have been auto-closed, my bad

**Comment by tyrasd** [comment]
> Add a confirmation step

An idea to improve this feature generally would be use the confirmation window to also offer the mapper's last changesets' locations to jump to. That would feel less like a  _"do you really want to do what you just clicked on"_ dialog that we want to avoid in general.

**Comment by doniks** [comment]
>  use the confirmation window 

Or maybe no window at all. Just a way to go to "my location" and also a way to go back to where I was/worked/back
